direct/client.go

```diff
@@ -0,0 +1,113 @@
+package direct
+
+import (
+	"context"
+	"sync"
+
+	"github.com/skycoin/skycoin/src/util/logging"
+
+	"github.com/skycoin/dmsg/cipher"
+	"github.com/skycoin/dmsg/disc"
+)
+
+var log = logging.MustGetLogger("direct")
+
+// APIClient implements dmsg discovery API client.
+type APIClient interface {
+	Entry(context.Context, cipher.PubKey) (*disc.Entry, error)
+	PostEntry(context.Context, *disc.Entry) error
+	PutEntry(context.Context, cipher.SecKey, *disc.Entry) error
+	DelEntry(context.Context, *disc.Entry) error
+	AvailableServers(context.Context) ([]*disc.Entry, error)
+}
+
+// directClient represents a client that doesnot communicates with a dmsg-discovery, instead directly gets the dmsg-server info via the user or is hardcoded, it
```

Comment:
It does communicate with the dmsg discovery but does so via bootstrapped dmsg connection. Or is direct client only respondible for the very first connection to the dmsg server?

Reply:
Direct client dose-not connect to the `dmsg-discovery` at all. It the `directClient` struct is a offline representation of the dmsg-discovery basically.